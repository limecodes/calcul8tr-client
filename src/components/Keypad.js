import React from 'react';

import { OPERATIONS } from '../constants';

import Button from './Button';

const {
    MULTIPLY, DIVIDE, SUBTRACT, ADD, EQUAL,
} = OPERATIONS;

const layout = [
  [7, 8, 9, DIVIDE],
  [4, 5, 6, MULTIPLY],
  [1, 2, 3, SUBTRACT],
  [0, EQUAL],
];

function Keypad(props) {
  return layout.flat().map((e, i) => (
    <div key={i} className="Button">
      <Button value={e} onPress={props.onPress} />
    </div>
  ));
}

export default Keypad;
