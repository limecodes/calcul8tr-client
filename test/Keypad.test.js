import React from 'react';
import renderer from 'react-test-renderer';
import Keypad from '../src/components/Keypad';

describe('Basic Keypad Functionality', () => {
  test('Renders correctly', () => {
    const tree = renderer.create(<Keypad />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
