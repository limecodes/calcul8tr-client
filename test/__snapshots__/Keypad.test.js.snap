// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Keypad Functionality Renders correctly 1`] = `
Array [
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={7}
    >
      7
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={8}
    >
      8
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={9}
    >
      9
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      className="operation"
      onClick={[Function]}
      value="DIVIDE"
    >
      ÷
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={4}
    >
      4
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={5}
    >
      5
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={6}
    >
      6
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      className="operation"
      onClick={[Function]}
      value="MULTIPLY"
    >
      ×
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={1}
    >
      1
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={2}
    >
      2
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={3}
    >
      3
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      className="operation"
      onClick={[Function]}
      value="SUBTRACT"
    >
      −
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      onClick={[Function]}
      value={0}
    >
      0
    </button>
  </div>,
  <div
    className="Button"
  >
    <button
      className="calculate"
      onClick={[Function]}
      value="EQUAL"
    >
      =
    </button>
  </div>,
]
`;
